### PR TITLE
Table component

### LINKED ISSUE #47: RFC: add in table component to new wui
as we work with a lot of tables in attorneys, i'm wondering whether that should be supported in here.

pros would be that it would be easy to make tables that match our design. it would be nice, if it's not considered too much, to add in  options to paginate and sort out of the box. i found myself basically working on creating this component within attorneys so that moving some things into tables is easier.

cons would be all the reasons we decided to clear out the old custom wui components.

opinions/thoughts?

### PR DESCRIPTION
**Pull request by kelarcrisp**
closes #47 

this should give us a pretty good starting point for a table component. look at `Table.stories.js` for an example of how to use it. happy to extend this further before this is released if anyone has suggestions

### CODE REVIEW

**Review by cviveiro — COMMENTED** [review]
thank you for this! this is all i see. looks like a good balance of providing a general component that will stay lightweight

File: src/stories/Table.stories.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  {
+    id: 5,
+    columnOne: 'test data 9',
+    columnTwo: 'test data 10',
+  },
+];
+
+const WuiTable = props => (
+  <Table {...props}>
+    <Table.Header headCells={headings.test_headings} />
```
[the remark this answers is not in the document]
Comment (kelarcrisp) [reply src/stories/Table.stories.js:44]: ah you aren't missing anything, when I set the headings obj up I was initially going to show off multiple tables in a loop so I did it that way. will fix

File: src/table/TableBody.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  return 0;
+};
+
+const customComparator = (reverse, orderColumnBy) =>
+  reverse
+    ? (a, b) => getInDescOrder(a, b, orderColumnBy)
+    : (a, b) => -getInDescOrder(a, b, orderColumnBy);
+
+const sortRecords = (tableRecords, comparator) => {
+  const tobeSortedRecords = tableRecords.map((record, i) => [record, i]);
```
[the remark this answers is not in the document]
Comment (kelarcrisp) [reply src/table/TableBody.js:28]: oops! fixed

File: src/table/TableHeader.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    <TableHead className={classes.tableHeader}>
+      <TableRow>
+        {headCells.map(headRow => (
+          <TableCell
+            align="left"
+            key={headRow.id}
+            // eslint-disable-next-line no-nested-ternary
+            sortDirection={orderBy === headRow.id ? (reverse ? 'desc' : 'asc') : false}
+          >
+            <TableSortLabel
```
[the remark this answers is not in the document]
Comment (kelarcrisp) [reply src/table/TableHeader.js:32]: yeah! we can definitely account for that scenario. I'll just simply add a `sort` key on the column headings for now as I think that should suffice. if there are more and more things like this that we want to add, I'll start to give some more thought to how we could handle these things

**Comment by kelarcrisp** [comment]
I also found a bug with the direction of the column arrow wasn't changing correctly, went ahead and fixed that as well

**Review by cviveiro — APPROVED** [review]
lgtm, thanks for doing this!! looking forward to the release so we can pull this into attys :) lmk when that's ready

**Pull request merged**